### ISSUE TITLE
 Questions about baseline and metalearn training 

### ISSUE THREAD

**Issue by hh23333** [cbfinn/maml#20]
Hello, Chelsea.

It seems that  when training the MAML on Omniglot and miniImagemet, there are meta_batch_size of tasks in one same batch. These tasks share the same network parameters, and each task has the labels from 0~4(5 ways) but classes between each tasks are not the same. My question is why is it make sense to do this.  
For the regression task, it seems that MAML method has 20 data points for training but baseline method only has 10 in one task: 
`self.pretrain_op = tf.train.AdamOptimizer(self.meta_lr).minimize(total_loss1)
 self.metatrain_op = optimizer.apply_gradients(gvs)`
MAML uses both inputa and inputb for training, but baseline method only used inputa.

**Comment by cbfinn** [comment]
> These tasks share the same network parameters, and each task has the labels from 0~4(5 ways) but classes between each tasks are not the same. My question is why is it make sense to do this.
Within a task, the classes are consistent. But, across tasks, the classes vary. This is the standard set-up for few-shot image recognition, as introduced by Santoro et al.

> For the regression task, it seems that MAML method has 20 data points for training but baseline method only has 10 in one task
Yes. I have found that the batch size (e.g. of 10 vs. 20) does not affect the performance of the oracle baseline.

**Comment by hh23333** [comment]
@cbfinn Thanks for your help

**Comment by michelleowen** [comment]
@cbfinn But this also makes the baseline without oracle only see 10 data points. shouldn't it see all 20 data points in order to compare with MAML apple-to-apple?
